up42/http/oauth.py

```diff
@@ -93,20 +93,21 @@ def token(self) -> Token:
         return self._token
 
 
-def detect_settings(credentials: dict) -> Optional[config.CredentialsSettings]:
-    if all(credentials.values()):
-        keys = credentials.keys()
-        if keys == {"project_id", "project_api_key"}:
-            warnings.warn(
-                "Project based authentication will be deprecated."
-                "Please follow authentication guidelines (/docs/authentication.md)."
-            )
-            return config.ProjectCredentialsSettings(**credentials)
-        if keys == {"username", "password"}:
-            return config.AccountCredentialsSettings(**credentials)
-        raise InvalidCredentials
-    elif any(credentials.values()):
-        raise IncompleteCredentials
+def detect_settings(credentials: Optional[dict]) -> Optional[config.CredentialsSettings]:
```

Comment:
are these changes coming from pre-commit hooks?
Is there a difference of version in your environment and poetry files?

Reply:
No, they are coming actual changes, config read from file can be None (I dropped or {} part since it also doesn't work with this code). Hence are the changes in the input signature and the actual code